WordPress/Classes/Services/ThemeService.m

```diff
@@ -0,0 +1,274 @@
+ */
+- (Theme *)newThemeWithId:(NSString *)themeId
+{
+    NSParameterAssert([themeId isKindOfClass:[NSString class]]);
+    
+    NSEntityDescription *entityDescription = [NSEntityDescription entityForName:[Theme entityName]
+                                                         inManagedObjectContext:self.managedObjectContext];
+    
+    __block Theme *theme = nil;
+    
+    [self.managedObjectContext performBlockAndWait:^{
+        theme = [[Theme alloc] initWithEntity:entityDescription
+               insertIntoManagedObjectContext:self.managedObjectContext];
+    }];
+    
+    return theme;
+}
+
+/**
+ *  @brief      Obtains the theme with the specified ID if it exists, otherwise a new theme is
+ *              created and returned.
+ *
+ *  @param      themeId     The ID of the theme to retrieve.  Cannot be nil.
+ *
+ *  @returns    The stored theme matching the specified ID if found, or nil if it's not found.
+ */
+- (Theme *)findOrCreateThemeWithId:(NSString *)themeId
+{
+    NSParameterAssert([themeId isKindOfClass:[NSString class]]);
+    
+    Theme *theme = [self findThemeWithId:themeId];
+    
+    if (!theme) {
+        theme = [self newThemeWithId:themeId];
+    }
+    
+    return theme;
+}
+
+#pragma mark - Local queries: finding themes
+
+- (Theme *)findThemeWithId:(NSString *)themeId
+{
+    NSParameterAssert([themeId isKindOfClass:[NSString class]]);
+    
+    Theme *theme = nil;
+    
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@""];
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:[Theme entityName]];
+    
+    fetchRequest.predicate = predicate;
+    
+    NSError *error = nil;
+    NSArray *results = [self.managedObjectContext executeFetchRequest:fetchRequest error:&error];
+    
+    if (results.count > 0) {
+        theme = (Theme *)[results firstObject];
+        NSAssert([theme isKindOfClass:[Theme class]],
+                 @"Expected a Theme object.");
+    } else {
+        NSAssert(error == nil,
+                 @"We shouldn't be getting errors here.  This means something's internally broken.");
+    }
+    
+    return theme;
+}
+
+#pragma mark - Remote queries: Getting theme info
+
+- (NSOperation *)getActiveThemeForBlog:(Blog *)blog
+                               success:(ThemeServiceThemeRequestSuccessBlock)success
+                               failure:(ThemeServiceFailureBlock)failure
+{
+    NSParameterAssert([blog isKindOfClass:[Blog class]]);
+    NSAssert([self blogSupportsThemeServices:blog],
+             @"Do not call this method on unsupported blogs, check with blogSupportsThemeServices first.");
+    
+    ThemeServiceRemote *remote = [[ThemeServiceRemote alloc] initWithApi:blog.restApi];
+    
+    NSOperation *operation = [remote getActiveThemeForBlogId:[blog dotComID]
+                                                     success:^(RemoteTheme *remoteTheme) {
+                                                         if (success) {
```

Comment:
We could just assert on the `success` callback, since without that one this call makes no sense.
But this is totally minor, and taste-y comment. Feel free to disregard!